docs/infrastructure/vs-deps-branches.md

```diff
@@ -0,0 +1,132 @@
+# Why do we have -vs-deps branches?
+
+The Roslyn repo includes projects that reference Visual Studio APIs. These projects create a challenge because we have two tensions pulling us in opposite directions:
+
+1. We often want this to be able to reference the latest Visual Studio APIs, even if the APIs are not yet shipped. If the editor team creates a new API for us to implement or consume, they would
+   like us to consume it as soon as they have merged their feature work into the (internal) Visual Studio master branch; this means they can get API feedback and shake out any bugs as soon as possible.
+2. We want public contributors to be able to contribute to all of Roslyn, the Visual Studio layers included. Even if the contributor isn't _changing_ the code in the Visual Studio layer, deploying to Visual
+   Studio is the only way to actually _test_ any changes made to any of our layers in the middle in an interactive way. A contributor making a change to a code fix or refactoring can of course write unit
+   tests, but they can't just "try it locally" to experiment.
+
+To find a balance between these two tensions, we have two branches. Using our master branch as an example, we have two branches named 'master' and 'master-vs-deps':
+
+1. The master branch as a rule must be buildable and runnable against the latest public non-preview build of Visual Studio. This means no change can be made inside of it that is dependent on an preview
```

Comment:
This is not correct; we have required uses to install previews for [all of dev17 development cycle](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md#developing-with-visual-studio-2022). Is this document stating the original reasons we created the *-vs-deps branches or is it expected to be correct in the present? Either way this should be clarified as we've apparently already has the conversation about allowing contributors to use VS Previews

Reply:
@jmarolf: this pull request is old and predates the Dev17 development cycle. It'll be closed if we go ahead with your proposal.

Reply:
(@allisonchou also just asked me about it, which makes me wonder what caused you both to notice it after two years!)

Reply:
For non-major releases of VS the statement is accurate. Our strong default is to avoid requiring previews to work on the `main` branch of the repository. That was part of the "easier it is to build, easier it is to contribute, more people we will get contributing- mentality. We deviated from this a handful of times over the years when the situation dictated it but those were exceptions.

For major releases of VS we have often deviated from this and required public previews. The reason being that there just weren't many other sane options on the table hence we required public previews. 

I would argue though that the Dev17 cycle is evidence that we've been a bit too cautious here. None of our contributors pushed back on our preview requirement that I am aware of. Makes it harder to argue this would limit contributions if we required preview releases.